ta4j-core/src/test/java/org/ta4j/core/num/NumTest.java

```diff
@@ -227,5 +235,72 @@ public void testArithmetic(){
         assertFalse(five.equals(five.function().apply((short)45)));
     }
 
-    //TODO: add precision tests for BigDecimalNum
+    @Test
+    public void sqrtOfBigInteger() {
+        String sqrtOfTwo = "1.4142135623730950488016887242096980785696718753769480731" +
+            "766797379907324784621070388503875343276415727350138462309122970249248360" +
+            "558507372126441214970999358314132226659275055927557999505011527820605715";
+
+        int precision = 200;
+        assertNumEquals(sqrtOfTwo, numOf(2).sqrt(precision));
+    }
+
+    @Test
+    public void sqrtOfBigDouble() {
+        String sqrtOfOnePointTwo = "1.095445115010332226913939565601604267905489389995966508453788899464986554245445467601716872327741252";
+
+        int precision = 100;
+        assertNumEquals(sqrtOfOnePointTwo, numOf(1.2).sqrt(precision));
+    }
+
+    @Test
+    public void sqrtOfNegativeDouble() {
+        assertTrue(numOf(-1.2).sqrt(12).isNaN());
+        assertTrue(numOf(-1.2).sqrt().isNaN());
+    }
+
+    @Test
+    public void sqrtOfZero() {
+        assertNumEquals(0, numOf(0).sqrt(12));
+        assertNumEquals(0, numOf(0).sqrt());
+    }
+
+    @Test
+    public void sqrtLudicrousPrecision() {
+        BigDecimal numBD = BigDecimal.valueOf(Double.MAX_VALUE).multiply(BigDecimal.valueOf(Double.MAX_VALUE).add(BigDecimal.ONE));
+        Num sqrt = numOf(numBD).sqrt(100000);
+        if (numOf(0).getClass().equals(DoubleNum.class)) {
+            assertEquals("Infinity", sqrt.toString());
+        } else if (numOf(0).getClass().equals(BigDecimalNum.class)) {
+            assertNumEquals("1.7976931348623157000000000000000E+308", sqrt);
+            assertNumNotEquals("1.7976931348623157000000000000001E+308", sqrt);
+            assertNumEquals(Double.MAX_VALUE, sqrt);
+            assertNumEquals(numOf(Double.MAX_VALUE), sqrt);
+            BigDecimal sqrtBD = new BigDecimal(sqrt.toString());
+            assertNumEquals(numOf(numBD), numOf(sqrtBD.multiply(sqrtBD, new MathContext(99999, RoundingMode.HALF_UP))));
+            assertNumEquals(numOf(numBD), sqrt.multipliedBy(sqrt));
+        } else if (numOf(0).getClass().equals(PrecisionNum.class)) {
+            Properties props = new Properties();
+            try (InputStream is = getClass().getResourceAsStream("numTest.properties")) {
+                props.load(is);
+                assertNumEquals(props.getProperty("sqrtCorrect100000"), sqrt);
+                assertNumNotEquals(props.getProperty("sqrtCorrect99999"), sqrt);
+                assertNumEquals(Double.MAX_VALUE, sqrt);
+                assertNumNotEquals(numOf(Double.MAX_VALUE), sqrt);
+                BigDecimal sqrtBD = new BigDecimal(sqrt.toString());
+                assertNumEquals(numOf(numBD), numOf(sqrtBD.multiply(sqrtBD, new MathContext(99999, RoundingMode.HALF_UP))));
+                assertNumNotEquals(numOf(numBD), sqrt.multipliedBy(sqrt));
+            } catch (IOException ioe) {
+                ioe.printStackTrace();
```

Comment:
Please let the test fail with a message if IOException has been thrown. I think you can call fail("no test properties found") to let a junit test fail